### PR TITLE
describe kubernetes-vX.Y.Z tag in README

### PR DESCRIPTION
**Pull request by caesarxuchao**
@lavalamp @mbohlool ptal. Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +124,12 @@ separate directories for each minor version. That soon proved to be a mistake.
 We are keeping the top-level folders in the 1.4 and 1.5 branches so that
 existing users won't be broken.
 
+### Where does the code come from
```
Comment (lavalamp) [inline README.md]: s/$/?$/

File: README.md [hunk 2]
```diff
@@ -123,6 +124,12 @@ separate directories for each minor version. That soon proved to be a mistake.
...
+
+client-go is synced from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go. Code changes are made in that location, merged into k8s.io/kubernetes and later synced here.
```
Comment (lavalamp) [inline README.md]: Add something about different branches?

File: README.md [hunk 3]
```diff
@@ -123,6 +124,12 @@ separate directories for each minor version. That soon proved to be a mistake.
...
+
+To help users who need to know from what version of kubernetes the client-go code is copied, we mark the commit that matches kubernetes `vX.Y.Z` tag with `kubernetes-vX.Y.Z` tag.
```
Comment (lavalamp) [inline README.md]: Does every kubernetes release result in a new tag in client-go?

Comment (caesarxuchao) [inline README.md]: Good question. Kubernetes makes patch releases for minor branch (e.g., release-1.6) every 2-4 weeks. As a minor branch gets old, the release frequency decreases. I think there will be too many tags if we match every Kubernetes tags. How about adding the `kubernetes-vX.Y.Z` tags on-request? cc @liggitt

Comment (liggitt) [inline README.md]: > How about adding the kubernetes-vX.Y.Z tags on-request?

-1 on ad hoc manual sync processes

**Comment by lavalamp** [comment]
I recommend tagging iff it resulted in changes to the client.

**Comment by mbohlool** [comment]
I just saw these set of PRs. If you want more attention from me next time, assign the PR to me so it shows up in my PR dashboard :)

**Comment by caesarxuchao** [comment]
> I recommend tagging iff it resulted in changes to the client.

I think we need to tag all k8s.io/* repos that are copied from staging together, so if a release changes any of such repos, we'll need to tag.

File: README.md (hunk 1 above)
Comment (caesarxuchao) [inline README.md]: Updated this line and other titles.

**Comment by caesarxuchao** [comment]
Updated. However, I don't know if it's a good idea to add the kubernetes-vX.Y.Z tags anymore, it involves periodic manual labor, and we might change the publishing procedure (e.g., remove staging/) anytime. @liggitt could you get me in contact with the engineers who request for the tag? I want to know if they can just use the timestamp of the commit to trace matching versions.

**Comment by lavalamp** [comment]
when (if?) we switch to developing in repos, this won't be a need any more,
because folks will be able to just look at what the main repo is vendoring
to see what versions are compatible.

**Pull request closed without merging**